src/Octoshift/Services/GithubApi.cs

```diff
@@ -884,7 +884,14 @@ public virtual async Task UpdateSecretScanningAlert(string org, string repo, int
 
         var url = $"{_apiUrl}/repos/{org.EscapeDataString()}/{repo.EscapeDataString()}/secret-scanning/alerts/{alertNumber}";
 
-        object payload = state == SecretScanningAlert.AlertStateOpen ? new { state } : new { state, resolution };
+        var payload = state == SecretScanningAlert.AlertStateOpen
+            ? (new { state })
+            : (object)(new
+            {
+                state,
+                resolution,
+                resolution_comment = resolutionComment ?? string.Empty
```

Comment:
Why is it being set to empty string? If nothing is passed it makes sense for the default to be `null`.
```suggestion
                resolution_comment = resolutionComment
```

Reply:
Hello, thanks for the comments. 
Added it that way to align with what is done for the dismissedComment for code scanning alerts (cf. https://github.com/github/gh-gei/blob/main/src/Octoshift/Services/GithubApi.cs#L933 )
```csharp
       var payload = state == "open"
            ? (new { state })
            : (object)(new
            {
                state,
                dismissed_reason = dismissedReason,
                dismissed_comment = dismissedComment ?? string.Empty
            });
        await _client.PatchAsync(url, payload);
```

Your suggestion makes sense though, so would you prefer me to do that in a separate pull request to align both changes, or change both it in this pull request?

Reply:
👋 ok got it. So I guess as you said the other one needs to be changed too. I think it's better to do that one in a separate PR.

Reply:
Understood. I have opened #1248 and added your suggestions there. Thanks again!